### DISCUSSION TITLE
Current Weather

### DISCUSSION THREAD

**Discussion by harshaMadiraju** [open-meteo/open-meteo#159] [Q&A]
As of now Current Weather returns only

  "current_weather": {
    "temperature": 30.0
    "windspeed": 2.6
    "winddirection": 164.0
    "weathercode": 2
    "time": "2022-11-25T12:00"
  }
  
  But, is there any way to get "apparent_temperature,dewpoint_2m,relativehumidity_2m,pressure_msl" as well for current weather. including them in the query params doesn't work

**Comment by patrick-zippenfenig** [comment]
Hi,

thanks for asking! I deliberately limited `current_weather` to a limited set weather variables. Many weather stations only measure those.

Data from `current_weather` and from `hourly` weather variables is consistent. You can just use the regular `hourly` variables and select the current hour. This way you can get all variables.

Best, Patrick

**Reply by harshaMadiraju** [reply]
Perfect. Using the same logic. Just wondering whether am missing something. Thank you for the reply though

**Reply by scadergit** [reply]
Follow-up question, does this mean the hourly "current hour" is observation data or forecast data?

**Reply by patrick-zippenfenig** [reply]
It can be both. For Europe and North America it is typically observation data, but it is based on a model analysis. This means, that measurement stations are combined with radar, satellite, etc into a regular grid of 1-5 kilometre grid spacing.

For other locations worldwide the current weather might be based on a forecast that is up to 6 hours old.

At some point, I might integrate measurement data directly to have a very short delay of 5-30 minutes. The drawbacks are: measurement stations have false readings, are not available everywhere and often do not have information about precipitation and clouds
